File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/inheritance/baseDerivedWithMultipleConstructor/AnimalMain.java
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.baseDerivedWithMultipleConstructor;

/**
 * Instantiation example for the baseAndDerivedClass example code
 *
 * @author Ranjith Krishnamurthy
 */
public class AnimalMain {
    public static void main(String[] args) {
        displayAnimal(new Cat());
        displayAnimal(new Horse());
        displayAnimal(new Dog());
    }

    private static void displayAnimal(Animal animal) {
        animal.displayDNA();
        animal.makeNoise();
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/inheritance/baseDerivedWithMultipleConstructor/AnimalMain.kt
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.baseDerivedWithMultipleConstructor

/**
 * Instantiation example for the baseAndDerivedClass example code
 *
 * @author Ranjith Krishnamurthy
 */
fun main() {
    displayAnimal(Cat())
    displayAnimal(Horse())
    displayAnimal(Dog())
}

private fun displayAnimal(animal: Animal) {
    animal.displayDNA()
    animal.makeNoise()
}